integration/kotlinx-coroutines-play-services/src/Tasks.kt

```diff
@@ -76,8 +96,26 @@ public fun <T> Task<T>.asDeferred(): Deferred<T> {
  * This suspending function is cancellable.
  * If the [Job] of the current coroutine is cancelled or completed while this suspending function is waiting, this function
  * stops waiting for the completion stage and immediately resumes with [CancellationException].
+ *
+ * For bi-directional cancellation, an overload that accepts [CancellationTokenSource] can be used.
  */
-public suspend fun <T> Task<T>.await(): T {
+public suspend fun <T> Task<T>.await(): T = awaitImpl(null)
+
+/**
+ * Awaits for completion of the task that is linked to the given [CancellationTokenSource] to control cancellation.
```

Comment:
```suggestion
 * Awaits the completion of the task that is linked to the given [CancellationTokenSource] to control cancellation.
```
The same goes for the `await` above.